extensions/kotlin-serialization/deployment/src/main/java/io/quarkus/kotlin/serialization/deployment/KotlinSerializationProcessor.java

```diff
@@ -0,0 +1,71 @@
+package io.quarkus.kotlin.serialization.deployment;
+
+import static io.quarkus.deployment.Feature.RESTEASY_REACTIVE_KOTLIN_SERIALIZATION;
+import static io.quarkus.deployment.annotations.ExecutionTime.STATIC_INIT;
+import static io.quarkus.resteasy.reactive.common.deployment.ServerDefaultProducesHandlerBuildItem.json;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.SyntheticBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.kotlin.serialization.KotSerMessageBodyReader;
+import io.quarkus.kotlin.serialization.KotSerMessageBodyWriter;
+import io.quarkus.kotlin.serialization.KotlinSerializationConfig;
+import io.quarkus.kotlin.serialization.SerializerRecorder;
+import io.quarkus.resteasy.reactive.common.deployment.ServerDefaultProducesHandlerBuildItem;
+import io.quarkus.resteasy.reactive.spi.MessageBodyReaderBuildItem;
+import io.quarkus.resteasy.reactive.spi.MessageBodyWriterBuildItem;
+import kotlinx.serialization.json.Json;
+
+public class KotlinSerializationProcessor {
+
+    @Inject
+    CombinedIndexBuildItem combinedIndexBuildItem;
```

Comment:
Please don't do this. Best add the type as method parameter.

Reply:
funny story.  that's actually an artifact of modeling this after the mongo-client processor.  It's not actually even used!  It's gone now.